types/redux-actions/index.d.ts

```diff
@@ -66,6 +67,10 @@ export type Reducer<State, Payload> = (state: State, action: Action<Payload>) =>
 
 export type ReducerMeta<State, Payload, Meta> = (state: State, action: ActionMeta<Payload, Meta>) => State;
 
+export type ReduxReducer<State, Payload> = (state: State | undefined, action: Action<Payload>) => State;
```

Comment:
Please elaborate on these changes, specifically why we can not just add `undefined` to a `Reducer` definition?

Reply:
@alexey-pelykh thanks for the quick review. The existing `Reducer` definition is used as `handleAction`'s parameter which is guaranteed to be passed in the state (because of the required `initialState`). I thought adding this extra reducer definition could better describe the library's API.

Reply:
@oddui you're welcome! Why not to use it as `Reducer<State | undefined, Payload>` where it can be undefined? Instead of adding extra type

Reply:
Hi @alexey-pelykh thanks for the suggestion. However that also sets the return type of the reducer to be `state | undefined`, which i don't think is correct.

Reply:
Valid point, then let's just make a better name than `ReduxReducer` vs `Reducer`, I'm not sure what's specific about Redux in the name.

Reply:
Happy to change that. Did you have any suggestions? Thanks

Reply:
Hi @alexey-pelykh does `ReduxCompatibleReducer` sound better?

Reply:
Yes, this works!

Reply:
@oddui please don't forget to address this ^ suggested name sounds good to me

Reply:
Oh yes. Just changed that. Please have a look @alexey-pelykh